plugin-server/functional_tests/ingestion.test.ts

```diff
@@ -317,108 +272,197 @@ test.concurrent('consumer updates timestamp exported to prometheus', async () =>
     }, 10_000)
 })
 
-// We only want to run these test if we are running with the delay all events
-// feature enabled. See https://github.com/PostHog/product-internal/pull/405 for
-// details.
-const testIfDelayEnabled = process.env.DELAY_ALL_EVENTS_FOR_TEAMS === '*' ? test.concurrent : test.concurrent.skip
-testIfDelayEnabled(
-    `anonymous event recieves same person_id if $identify happenes shortly after, and there's already an anonymous person`,
-    async () => {
-        const teamId = await createTeam(postgres, organizationId)
-        const initialDistinctId = new UUIDT().toString()
-        const secondDistinctId = new UUIDT().toString()
-        const personIdentifier = new UUIDT().toString()
-
-        // First we emit an anoymous event and wait for the person to be
-        // created.
-        const initialEventId = new UUIDT().toString()
-        await capture(producer, teamId, initialDistinctId, initialEventId, 'custom event')
-        await waitForExpect(async () => {
-            const persons = await fetchPersons(clickHouseClient, teamId)
-            expect(persons).toContainEqual(
+test.concurrent(`event ingestion: initial login flow keeps the same person_id`, async () => {
+    const teamId = await createTeam(postgres, organizationId)
+    const initialDistinctId = 'initialDistinctId'
+    const personIdentifier = 'test@posthog.com'
+
+    // This simulates initial sign-up flow,
+    // where the user has first been browsing the site anonymously for a while
+
+    // First we emit an anoymous event and wait for the person to be
+    // created.
+    const initialEventId = new UUIDT().toString()
+    await capture(producer, teamId, initialDistinctId, initialEventId, 'custom event')
+    await waitForExpect(async () => {
+        const persons = await fetchPersons(clickHouseClient, teamId)
+        expect(persons).toContainEqual(
+            expect.objectContaining({
+                properties: expect.objectContaining({ $creator_event_uuid: initialEventId }),
+            })
+        )
+    }, 10000)
+
+    // We then identify the person
+    await capture(producer, teamId, personIdentifier, new UUIDT().toString(), '$identify', {
+        distinct_id: personIdentifier,
+        $anon_distinct_id: initialDistinctId,
+    })
+
+    await waitForExpect(async () => {
+        const events = await fetchEvents(clickHouseClient, teamId)
+        expect(events.length).toBe(2)
+        expect(new Set(events.map((event) => event.person_id)).size).toBe(1)
+    }, 10000)
+})
+
+const testIfPoEEmbraceJoinEnabled = process.env.POE_EMBRACE_JOIN_FOR_TEAMS === '*'
+test.concurrent(`single merge results in an entry in person overrides`, async () => {
```

Comment:
I was thinking about rather than explicitly look for overrides, which is an implementation detail and the user shouldn't care about this, can we do the join on the overrides in `fetchEvents` and assert the user facing functionality we'd want. That should mean we have cases tested no matter if we change the implementation. What do you think?

Reply:
taking this a step further instead of a join we might want to use the dictionaries from https://github.com/PostHog/posthog/pull/13946 that @tomasfarias  added, but I'm not sure how I can best do that given the potential update delay, any ideas?

Reply:
Hmm yeah, that's an annoying one. If we could make the dict update shorter in these tests that would be handy.

Reply:
Just for testing purposes, I feel there are two options:
* Running a `SYSTEM RELOAD DICTIONARY` which forces an update to assert the dictionary is populated with the right values and we get the expected user facing functionality.
* Creating the dictionary with a significantly lower LIFETIME parameter for testing purposes, and waiting for this short lifetime to trigger an update.

The first one is easier as we can force an update, but it's a bit unrealistic. The second one requires us to parametrize the query and add some retrying while we wait for the lifetime, so a bit more setup but potentially more representative of the real case.

Reply:
In my PR the dictionary is created with a lifetime of 5 to 10 seconds, so maybe we wouldn't even need to shorten that as worst case we wouldn't be waiting more than 10 seconds (and perhaps a bit more due to lag).

Reply:
@hazzadous Forcing manual dictionary updates, at least from a client side perspective. Updates should either be automatic from ClickHouse (via LIFETIME) or perhaps tied to the squashing job.

Reply:
@tomasfarias yup makes sense.